src/app_service/service/chat/service.nim

```diff
@@ -94,7 +95,20 @@ QtObject:
     result.contactService = contactService
     result.chats = initTable[string, ChatDto]()
 
+  proc updateOrAddChat*(self: Service, chat: ChatDto) =
```

Comment:
I'd just do a forward declaration instead of moving it above init. IMO, init, setup, newService and delete should all be the first functions for easier access, but then you can put
```nim
# Forward declarations
proc updateOrAddChat*(self: Service, chat: ChatDto)
```
above them and it's good